code/game/objects/items/rcd/RPD.dm

```diff
@@ -414,23 +442,17 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 	var/atom/attack_target = atom_to_attack
 
 	//So that changing the menu settings doesn't affect the pipes already being built.
-	var/queued_p_type = recipe.id
-	var/queued_p_dir = p_dir
-	var/queued_p_flipped = p_flipped
+	var/queued_pipe_type = recipe.id
+	var/queued_pipe_dir = p_dir
+	var/queued_pipe_flipped = p_flipped
 
 	//Unwrench pipe before we build one over/paint it, but only if we're not already running a do_after on it already to prevent a potential runtime.
 	if((mode & DESTROY_MODE) && (upgrade_flags & RPD_UPGRADE_UNWRENCH) && istype(attack_target, /obj/machinery/atmospherics) && !(DOING_INTERACTION_WITH_TARGET(user, attack_target)))
 		attack_target = attack_target.wrench_act(user, src)
 		if(!isatom(attack_target)) //can return null, FALSE if do_after() fails see /obj/machinery/atmospherics/wrench_act()
 			return TRUE
 
-	//make sure what we're clicking is valid for the current category
-	var/static/list/make_pipe_whitelist
-	if(!make_pipe_whitelist)
-		make_pipe_whitelist = typecacheof(list(/obj/structure/lattice, /obj/structure/girder, /obj/item/pipe, /obj/structure/window, /obj/structure/grille))
-	if(istype(attack_target, /obj/machinery/atmospherics) && mode & BUILD_MODE)
-		attack_target = get_turf(attack_target)
```

Comment:
The code below relies on attack_target being modified, clicking on a pipe with all modes enabled would previously try to build another pipe, now it tries to reconfigure the pipe